lib/tests.js

```diff
@@ -0,0 +1,361 @@
+     */
+    function requeue(callback) {
+        console.log("Starting requeue test");
+        runOnMessage((err, data) => {
+            if (err) {
+                callback(err);
+            }
+
+            data.requeue((rqErr) => {
+                instance.removeAllListeners("data");
+                callback(rqErr);
+            });
+        });
+    }
+
+
+    /**
+     * Runs heartbeat test.
+     *
+     * @param {Function} callback
+     */
+    function heartbeat(callback) {
+        console.log("Starting heartbeat test");
+        runOnMessage((err, data) => {
+            if (err) {
+                callback(err);
+            }
+
+            data.heartbeat((rqErr) => {
+                instance.removeAllListeners("data");
+                callback(rqErr);
+            });
+        });
+    }
+
+
+    /**
+     *  Removes test message.
+     *
+     * @param {Function} callback
+     */
+    function remove(callback) {
+        console.log("Starting removal test");
+        runOnMessage((err, data) => {
+            if (err) {
+                callback(err);
+            }
+
+            data.remove((rqErr) => {
+                instance.removeAllListeners();
+                callback(rqErr);
+            });
+        });
+    }
+
+
+    /**
+     * Tests to ensure multiple listeners don't get the same message
+     * and that the locking mechanism is in fact working.
+     *
+     * @param {Function} callback
+     */
+    function multipleListeners(callback) {
+        var accessed, instanceTwo;
+
+        console.log("Setting multiple listeners for test.");
+        accessed = false;
+        config.owner = "testerTwo";
+        instanceTwo = new Plugin(config);
+
+
+        /**
+         * Deals with data returned by either listeners.
+         *
+         * @param {Object} data
+         */
+        function dealWithData(data) {
+            console.log(`Retrieved message on mutliple listener test ${data.message}`);
+            if (!accessed && data.message === "multiMsg") {
+                accessed = true;
+            } else if (accessed && data.message === "multiMsg") {
+                instance.removeAllListeners();
+                instanceTwo.removeAllListeners();
+                callback(new Error("Same message processed by both listeners! Fail!"));
+            }
+        }
+
+        instanceTwo.connect((conErr) => {
+            if (conErr) {
+                console.log("Error connecting to second instance.");
+                instanceTwo.removeAllListeners();
+                callback(conErr);
+
+                return;
+            }
+
+            instance.on("data", (data) => {
+                console.log(`First listener accessed message ${data.message}`);
+                dealWithData(data);
+
+                // Making sure same message isn't access twice until requeued.
+                // Giving it a somewhat arbitrary amount of time.
+                setTimeout(() => {
```

Comment:
Yea I did that because I listen on one first so it would always get it first. I guess I shouldn't rely on that. Fixed